### PR TITLE
Feature: Add Extended Reality key concepts and topics

### LINKED ISSUE #6: Extended Reality (XR) Fundamentals
**Is your feature request related to a problem? Please describe.**
This issue aims to add base documentation files for Extended Reality (XR) fundamental areas, background and concepts.

**Describe the Solution You'd Like**
 - [x] Definition of Extended Reality (XR)
 - [x] Extended Reality Areas
 - [x] Understanding User Interaction
 - [x] Importance of Extended Reality (XR) and it's current state of development in the market
 - [x] Extended Reality Programs and background

**Additional Context**
 - #3

### PR DESCRIPTION
**Pull request by Domenic-MZS**
# Feature: Add Extended Reality key concepts and topics
Closes #6.

## Overview:
 - Issue: Lack of information on Extended Reality key concepts and topics.
 - Context: The project currently lacks documentation on extended reality. Establishing initial fundamentals, defining key concepts, exploring XR types, usages, limitations, and contextualizing its relevance is pivotal at this stage.
 - Solutions/Dependencies: Consolidated information sourced from multiple reputable sites, identifying common patterns and streamlining crucial aspects, sections, and areas.
 - Impact: This contributes significantly to the project by providing easily accessible and valuable insights into extended reality. It enhances user experience and knowledge acquisition while navigating this repository for extended reality information. Understanding fundamental concepts and pathways within XR becomes more accessible.

## Description
This pull request addresses the absence of comprehensive information on Extended Reality (XR) within the project repository. Recognizing the significance of establishing a foundational understanding of XR technologies, this contribution aims to lay out fundamental concepts, key areas, usage scenarios, limitations, and the current landscape of XR. By consolidating information from various reputable sources and distilling it into a concise yet comprehensive format, this addition seeks to empower users with accessible knowledge to navigate and comprehend the realm of Extended Reality effectively.

## Changes Made
Include an extended XR Readme, covering essential elements like the definition of Extended Reality (XR), areas within XR, insights into user interaction, the significance of XR and its current market status, as well as background information on XR programs. Additionally, foundational documentation files were introduced, addressing crucial aspects of XR, and various pertinent sections were added to the table of contents, encompassing definitions, importance, types, current state, user interaction, business applications, and challenges and limitations within the XR domain. 

There's also a new script to change markdown flavors accordingly to the selected environment (Obsidian or GitHub)

Content Relation Graph
> Note that the `README` from the middle, corresponds to the `docs/README.md` (Extended Reality), wich then connects or relates to the other topics or nodes in the image. It may need to change file names to adjust the graph labels, or find another alternative like a shell script to do that.

## Notes and References
 - [Obsidian Path compatibility](https://forum.obsidian.md/t/support-img-and-video-tag-with-src-relative-path-format/18647)
 - [Obsidian Heading link compatibility](https://forum.obsidian.md/t/heading-link-compatibility/46988)
 - [Change Nodes name on Obsidian Graph](https://forum.obsidian.md/t/use-h1-or-yaml-property-title-instead-of-or-in-addition-to-filename-as-display-name/687)
 - General Idea: #3 
 - Issue: #6 
 - [Extra requirements for accessibility](https://github.com/Domenic-MZS/Essential-ARscape/issues/6#issuecomment-1837036134)

### CODE REVIEW

**Comment by Domenic-MZS** [comment]
Sorry for the force-push; I noticed the wrong git tree hierarchy too late.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+- [Definition](docs/README.md#What%20really%20is%20the%20Extended%20Reality%20(XR)?)
+- [Importance](docs/README.md#Why%20is%20it%20so%20important?)
+- [Types of Extended Reality](docs/README.md#Types%20of%20Extended%20Reality)
+- [Current State](<docs/README.md#Current XR State>)
+- [User Interaction](/docs/README.md#Extended%20Reality%20(XR)%20User%20Interaction)
+- [Business Applications](docs/README.md#Extended%20Reality%20Applications%20for%20Business)
+- [Challenges and Limitations](<docs/README.md#Current challenges and limitations of Extended Reality>)
+
+### What really is the Extended Reality (XR)?
+Extended Reality is a catch-all "umbrella" term used to describe [immersive technologies](https://www.sciencedirect.com/topics/psychology/immersive-technology) that can merge the physical and virtual world, including [Augmented Reality (AR)](docs/augmented-reality/README.md), [Virtual Reality](docs/virtual-reality/README.md), and [Mixed Reality](docs/mixed-reality/README.md) plus those that are still to be created.
```
Comment (Domenic-MZS) [inline docs/README.md:12]: Same issue of links path `/`

**Review by Domenic-MZS — COMMENTED** [review]
The are a bunch of broken links and possible connections for sections on Ar/Vr/Mr

I also think the abbrevs should be used at the start too, at least in ()

**Comment by Domenic-MZS** [comment]
Hi @Mobinake, feel free to review this Pull Request (PR) to `Rebase and merge` or `Squash Commit` those changes into main. Any feedback is welcome! 👾 🦖

**Review by Mobinake — APPROVED** [review]

**Review by Mobinake — APPROVED** [review]

**Review by Mobinake — APPROVED** [review]

**Review by Mobinake — APPROVED** [review]
good

**Pull request merged**